### DISCUSSION TITLE
It would be useful to be able to view and  edit XML files

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#7153] [Editor]
XML is an extensively used file format and allows more detailed markup than you can get in systems like JSON or YAML.  It would be great if Godot's editor could list them in the resource browser and edit them in the text editor.

**Comment by theraot** [comment]
Currently Godot has API for parsing XML and JSON (there is no core support for YAML), so it makes sense to include them as text resources. And some basic syntax highlighting should be possible too.

**Reply by Calinou** [reply]
See https://github.com/godotengine/godot/pull/77972 and https://github.com/godotengine/godot/pull/78312 for inspiration. Note that `*.xml` will also have to be added to the `textfile_extensions` editor setting's default value.

Feel free to work on this if you have time, as I won't tackle this in the near future.
